manifests/repo/apt.pp

```diff
@@ -0,0 +1,103 @@
+#
+# == Class gluster::repo::apt
+#
+# enable the upstream Gluster Apt repo
+#
+# === Parameters
+#
+# version: the version to use when building the repo URL
+# repo_key_name: The repo signing key or fingerprint
+# repo_key_path: ignored
+# repo_key_source: ignored
+# priority: Apt pin priority to set for the Gluster repo
+#
+# Currently only released versions are supported.  If you want to use
+# QA releases or pre-releases, you'll need to edit line 54 below
+
+# === Examples
+#
+# Enable the Apt repo, and use the public key supplied
+#
+# class { gluster::repo::yum:
+#   repo_key_name => 'F7C73FCC930AC9F83B387A5613E01B7B3FE869A9',
+# }
+#
+# === Authors
+#
+# Drew Gibson <dgibson@rlsolutions.com>
+#
+# === Copyright
+#
+# Copyright 2015 RL Solutions, unless otherwise noted
+#
+class gluster::repo::apt (
+  $version         = $::gluster::params::version,
+  $repo_key_name   = $::gluster::params::repo_gpg_key_name,
+  $repo_key_path   = $::gluster::params::repo_gpg_key_path,
+  $repo_key_source = $::gluster::params::repo_gpg_key_source,
+  $priority        = $::gluster::params::repo_priority,
+) {
+  include 'apt'
+  
+  # basic sanity check
+  if $version == 'LATEST' and $::operatingsystem != 'Ubuntu' {
+    $repo_ver = $version
+  } else {
+    if $version =~ /^\d\.\d$/ {
+      $repo_ver = $::operatingsystem ? {
+        'Ubuntu' => "${version}",
+        default  => "${version}/LATEST",
+      }
+    } elsif $version =~ /^(\d)\.(\d)\.(\d).*$/ {
+      $repo_ver = $::operatingsystem ? {
+        'Ubuntu' => "${1}.${2}",
+        default  => "${1}.${2}/${1}.${2}.${3}",
+      }
+    } else {
+      fail("${version} doesn't make sense for $::operatingsystem!")
+    }
+  }
+    
+  # the Gluster repo only supports x86_64 and i386. The Ubuntu PPA also supports armhf and arm64.
+  case $::operatingsystem {
+    'Debian': {
+      $arch = $::architecture ? {
+        'amd64' => 'amd64',
+        /i\d86/ => 'i386',
+        default => false,
+      }
+      $repo_base = 'https://download.gluster.org/pub/gluster/glusterfs'
+      $repo_url = "${repo_base}/${repo_ver}/Debian/${::lsbdistcodename}/apt/"
+    }
+    default: {
+      $arch = $::architecture ? {
+        'amd64'      => 'amd64',
+        /i\d86/      => 'i386',
+        /armv[67].*/ => 'armhf',
+        /armv8.*/    => 'arm64',
+        default      => false,
+      }
+      $repo_url = "http://ppa.launchpad.net/gluster/glusterfs-${repo_ver}/ubuntu/"   
```

Comment:
The PPA _is_ the upstream repo for Ubuntu. There is a PPA resource in pupputlabs apt module but I used apt::source for consistency.

I'm having some difficulties with the Debian and Ubuntu public keys not found. They are on the keyservers so I have to read up and troubleshoot.

Drew

Sent from my phone

> On Dec 3, 2015, at 10:07 AM, Scott Merrill <notifications@github.com> wrote:
> 
> In manifests/repo/apt.pp:
> 
> > +        'amd64' => 'amd64',
> > +        /i\d86/ => 'i386',
> > +        default => false,
> > +      }
> > +      $repo_base = 'https://download.gluster.org/pub/gluster/glusterfs'
> > +      $repo_url = "${repo_base}/${repo_ver}/Debian/${::lsbdistcodename}/apt/"
> > +    }
> > +    default: {
> > +      $arch = $::architecture ? {
> > +        'amd64'      => 'amd64',
> > +        /i\d86/      => 'i386',
> > +        /armv[67].*/ => 'armhf',
> > +        /armv8.*/    => 'arm64',
> > +        default      => false,
> > +      }
> > +      $repo_url = "http://ppa.launchpad.net/gluster/glusterfs-${repo_ver}/ubuntu/"   
> Should there be an option for Ubuntu users to use the PPA versus the upstream repo? If I read this correctly, Ubuntu users will always use the PPA?
> 
> —
> Reply to this email directly or view it on GitHub.
>